Source/ACE/Entity/Player.cs

```diff
@@ -1095,32 +1095,43 @@ public void HandleActionBuy(ObjectGuid vendorId, List<ItemProfile> items)
         /// <param name="purchaselist"></param>
         /// <param name="valid"></param>
         /// <param name="goldcost"></param>
-        public void HandleActionBuyFinalTransaction(WorldObject vendor, List<WorldObject> purchaselist, bool valid, uint goldcost)
+        public void HandleActionBuyFinalTransaction(WorldObject vendor, List<WorldObject> uqlist, List<WorldObject> genlist, bool valid, uint goldcost)
         {
             new ActionChain(this, () =>
             {
+                // to do process payment - its all free!
                 // vendor accepted the transaction
                 if (valid)
                 {
-                    ////if (SpendCoin(goldcost))
-                    ////{  // player is has enough cash to buy items
-                    ////    foreach (WorldObject wo in purchaselist)
-                    ////    {
-                    ////        wo.ContainerId = Guid.Full;
-                    ////        wo.Placement = 0;
-                    ////        AddToInventory(wo);
-                    ////        TrackObject(wo);
-                    ////    }
-                    ////}
-                    ////else // not enough cash.
-                    ////    valid = false;
+                    foreach (WorldObject wo in uqlist)
+                    {
+                        wo.ContainerId = Guid.Full;
+                        wo.Placement = 0;
+                        AddToInventory(wo);
+                        Session.Network.EnqueueSend(new GameMessagePutObjectInContainer(Session, Guid, wo, 0));
+                        Session.Network.EnqueueSend(new GameMessageUpdateInstanceId(Guid, wo.Guid, PropertyInstanceId.Container));
+                    }
+                    // these are new items genated by vendor.
+                    foreach (WorldObject wo in genlist)
+                    {
+                        wo.ContainerId = Guid.Full;
+                        wo.Placement = 0;
+                        AddToInventory(wo);
+                        Session.Network.EnqueueSend(new GameMessageCreateObject(wo));
+                        if (wo.WeenieType == WeenieType.Container)
+                            Session.Network.EnqueueSend(new GameEventViewContents(Session, wo.SnapShotOfAceObject()));
```

Comment:
SnapShotting objects is for creating a point-in-time copy to save.  Are you sure that's what needs to be used, here?